main.tf

```diff
@@ -230,6 +238,13 @@ resource "google_compute_subnetwork" "traffic" {
   ip_cidr_range = var.traffic_subnet_cidr
 }
 
+resource "google_compute_network" "protected" {
```

Comment:
I'm worried that this gets created every time for every vendor even if they didn't want a protected network. Is that true?

Reply:
That's a good point, let me verify the Terraform output once.

Reply:
Thanks for pointing this out, I had missed adding for_each in the network interface resource creation block. This should make sure the resource only gets created when add_protected_network variable is true.